visualization/dashboard/src/MLIDashboard/Controls/DatasetExplorerTab/DatasetExplorerTab.tsx

```diff
@@ -0,0 +1,614 @@
+            return dict;
+        });
+        if (chartProps.chartType === ChartTypes.Scatter) {
+            const xAxis = chartProps.xAxis;
+            if (xAxis && xAxis.property && xAxis.options.dither) {
+                const rawX = cohort.unwrap(chartProps.xAxis.property);
+                rawX.forEach((val, index) => {
+                    // If categorical, show string value in tooltip
+                    if (jointData.metaDict[chartProps.xAxis.property].isCategorical) {
+                        customdata[index]["X"] = jointData.metaDict[chartProps.xAxis.property]
+                            .sortedCategoricalValues[val];
+                    } else {
+                        customdata[index]["X"] = val;
+                    }
+                });
+            }
+            const yAxis = chartProps.yAxis;
+            if (yAxis && yAxis.property && yAxis.options.dither) {
+                const rawY = cohort.unwrap(chartProps.yAxis.property);
+                rawY.forEach((val, index) => {
+                    // If categorical, show string value in tooltip
+                    if (jointData.metaDict[chartProps.yAxis.property].isCategorical) {
+                        customdata[index]["Y"] = jointData.metaDict[chartProps.yAxis.property]
+                            .sortedCategoricalValues[val];
+                    } else {
+                        customdata[index]["Y"] = val;
+                    }
+                });
+            }
+        }
+        return customdata;
+    }
+
+    private generateDefaultChartAxes(): void {
+        let maxIndex: number = 0;
+        let maxVal: number = Number.MIN_SAFE_INTEGER;
+        const yKey = JointDataset.DataLabelRoot + maxIndex.toString();
+        const yIsDithered = this.props.jointDataset.metaDict[yKey].isCategorical;
+        const chartProps: IGenericChartProps = {
+            chartType: ChartTypes.Scatter,
+            xAxis: {
+                property: JointDataset.IndexLabel,
+                options: {}
+            },
+            yAxis: {
+                property: yKey,
+                options: {
+                    dither: yIsDithered,
+                    bin: false
+                }
+            },
+            colorAxis: {
+                property: this.props.jointDataset.hasPredictedY ?
+                    JointDataset.PredictedYLabel : JointDataset.IndexLabel,
+                options: {}
+            }
+        }
+        this.props.onChange(chartProps);
+    }
+
+    private scatterSelection(guid: string, selections: string[], plotlyProps: IPlotlyProperty): void {
+        const selectedPoints =
+            selections.length === 0
+                ? null
+                : plotlyProps.data.map(trace => {
+                      const selectedIndexes: number[] = [];
+                      if ((trace as any).customdata) {
+                          ((trace as any).customdata as any[]).forEach((dict, index) => {
```

Comment:
(minor) can this be assigned to variable:
(trace as any).customdata
then it won't need to be redone twice